graph.go

```diff
@@ -171,11 +171,13 @@ func New[K comparable, T any](hash Hash[K, T], options ...func(*Traits)) Graph[K
 		option(&p)
 	}
 
+	s := newMemoryStore[K, T]()
+
 	if p.IsDirected {
-		return newDirected(hash, &p)
+		return newDirected(hash, &p, s)
 	}
 
-	return newUndirected(hash, &p)
+	return newUndirected(hash, &p, s)
 }
```

Comment:
There also should be a function`NewWithStore` for creating a graph with a custom `Store`.

Reply:
Added `NewWithStore`